lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,455 @@
+        #        This gives a decreasing influence on the extrapolated lightning
+        #        nowcast over forecast_period while retaining an influence from
+        #        the 50 km halo.
+
+        self.lrt_lev1 = lambda mins: 0.5 + mins * 2. / 360.
+        # Lightning-rate threshold for Lightning Risk 2 level
+        #        lrt_lev2 is the lightning rate threshold (as float) for
+        #        increasing first-guess lightning probability to risk 2 (LR2).
+        self.lrt_lev2 = 0.
+        # Prob(lightning) value for Lightning Risk 1 & 2 levels
+        #    pl_dict (dict):
+        #        Lightning probability values to increase first-guess to if
+        #        the lightning_thresholds are exceeded in the nowcast data.
+        #        Dict must have keys 1 and 2 and contain float values.
+        #        The default values are selected to represent lightning risk
+        #        index values of 1 and 2 relating to the key.
+
+        self.pl_dict = {1: 1., 2: 0.25}
+
+        # Set values for handling precipitation rate data
+        #    precipthr (tuple):
+        #        Values for limiting prob(lightning) with prob(precip).
+        #        These are the three prob(precip) thresholds and are designed
+        #        to prevent a large probability of lightning being output if
+        #        the probability of precipitation is very low.
+        self.precipthr = (0.0, 0.05, 0.1)
+        #    ltngthr (tuple):
+        #        Values for limiting prob(lightning) with prob(precip)
+        #        These are the three prob(lightning) values to scale to.
+        self.ltngthr = (0.0067, 0.25, 1.)
+        #    probability thresholds for increasing the prob(lightning)
+        #        phighthresh for heavy precip (>7mm/hr)
+        #            relates to problightning_values[2]
+        #        ptorrthresh for intense precip (>35mm/hr)
+        #            relates to problightning_values[1]
+        self.phighthresh = 0.4
+        self.ptorrthresh = 0.2
+        #    pl_dict (dict):
+        #        Lightning probability values to increase first-guess to if
+        #        the lightning_thresholds are exceeded in the nowcast data.
+        #        Dict must have keys 1 and 2 and contain float values.
+        #        The default values are selected to represent lightning risk
+        #        index values of 1 and 2 relating to the key.
+        self.pl_dict = {1: 1., 2: 0.25}
+
+        # Set values for handling vertically-integrated-ice (VII) data
+        #    ice_thresholds (tuple):
+        #        Values for increasing prob(lightning) with column-ice data.
+        #        These are the three VII thresholds in kg/m2.
+        self.ice_thresholds = (0.5, 1.0, 2.0)
+        #    ice_scaling (tuple):
+        #        Values for increasing prob(lightning) with VII data.
+        #        These are the three prob(lightning) values to scale to.
+        self.ice_scaling = (0.1, 0.5, 0.9)
```

Comment:
The information in these comments is very useful.  It might be better to move the bulk of the text to the docstring, and to standardise the formatting of those retained as inline comments (they're not args any more!).

Reply:
I think I prefer these how they are. The doc-string already includes a user-readable description of what is going to happen. These in-line comments describe which definition controls what. I agree that it was lazy of me to leave them looking like Args, but I think this still contains useful information in this form and I think it is easier to read like this.